### PR TITLE
Deal with NULL `GC$NonLinearModel`

### PR DESCRIPTION
**Pull request by CeresBarros**
Checking for class error  is not sufficient in cases where `is.null(GC$NonLinearModel)` -- I believe these come from lack of data, but perhaps need to dig further

In my fix, I check for `is.null(GC$NonLinearModel)` and also print a message differently when this is the case to avoid printing the full GC object

### CODE REVIEW

File: R/aNPPfunctions.R [hunk 1]
```diff
@@ -445,9 +445,14 @@ editSpeciesTraits <- function(name, GC, traits, fT, fB, speciesEquiv, sppCol, ma
     ## catch when not all models converged
     classesNonLinear <- unlist(lapply(GC$NonLinearModel, class))
     ## decided to allow non-converged GCs, if non-linear converged
-    if (any("try-error" %in% c(classesNonLinear))) {
+    if (any("try-error" %in% c(classesNonLinear)) | is.null(classesNonLinear)) {
```
Comment (achubaty) [inline R/aNPPfunctions.R]: Needs `||` for conditional

**Review by achubaty — CHANGES_REQUESTED** [review]

File: R/aNPPfunctions.R (hunk 1 above)
Comment (CeresBarros) [reply R/aNPPfunctions.R]: Thanks! Done

File: Biomass_speciesParameters.R [hunk 2]
```diff
@@ -16,10 +16,9 @@ defineModule(sim, list(
                    before = c("Biomass_core")),
   reqdPkgs = list("crayon", "data.table", "disk.frame", "fpCompare", "ggplot2", "gridExtra",
                   "magrittr", "mgcv", "nlme", "purrr", "robustbase", "sf",
+                  "reproducible (>= 1.2.10.9001)", "SpaDES.core (>= 2.0.2.9004)",
```
Comment (achubaty) [inline Biomass_speciesParameters.R]: can we set these to v 2.1.0 or higher, which are/will be the current CRAN versions?

Comment (achubaty) [reply Biomass_speciesParameters.R]: (reproducible < v2.0.0 most likely won't work -- too many breaking changes going to v2)

**Pull request merged**